telosnetwork/telos-wallet
# #367 | NFT/Collectible inventory page

# Fixes #367 

## Description
This PR adds live data and design implementation to the NFT inventory page. note that filters are to be implemented in a separate ticket

design: https://www.figma.com/file/eIpbnt6vqwqptEbZWtwO8Y/v0.3?node-id=133%3A1619

other changes:
- 'play' button icon changed to 'movie' icon for video NFT preview to be less ambiguous (as play button also applies to audio NFTs)
- long NFT names truncated with ellipse for NFT tiles
- for NFTs which contain their ID within the name, which is very common, ID is now hidden on NFT preview tile to prevent redundant info
- initial fetching of NFTs for user moved from NFT store init function to NFT inventory component to eliminate unnecessary network traffic
- corrected font size for ExternalLink component

## Test scenarios
Note: this will need to be tested locally if you, like me, do not have NFTs on your account
- run `git fetch --all && git checkout 367-nft-inventory-page` from telos wallet repo directory
- go into `nfts.ts` and edit line 66 to be `const owner = '0x13B745FC35b0BAC9bab9fD20B7C9f46668232607';` (an account with a lot of NFTs)
- go to `.env` and edit it to say `NETWORK="mainnet"`
- run `yarn dev`
- go to http://localhost:8081
- log in
- go to http://localhost:8081/evm/collectible-inventory
    - you should see a sensible loading state
    - NFTs should load
- use the grid/list toggle to change view
- reload the page
    - your grid/list preference should persist across reload
- look at NFTs in both grid and tile view
    - they should match designs
    - NFTs with no image should have a sensible icon displayed
    - long IDs should be truncated
- move around to different pages of NFTs
    - the URL should change to reflect the table state
    - pagination functions should all work as expected
- copy the url including pagination query params
- open new tab
- paste and go
    - the inventory page should load from deeplink to the correct page with the correct number of results
- switch to list view and click on the preview/icon of an NFT
    - you should be taken to the correct detail page 

## Checklist:
